plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/core/startup/PluginCoreEnforcementActivity.kt

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.startup
+
+import com.intellij.ide.ApplicationInitializedListener
+import com.intellij.ide.plugins.PluginEnabler
+import com.intellij.ide.plugins.PluginManagerCore
+import com.intellij.openapi.application.ex.ApplicationManagerEx
+import com.intellij.openapi.extensions.PluginId
+import com.intellij.openapi.progress.EmptyProgressIndicator
+import kotlinx.coroutines.CoroutineScope
+import org.slf4j.LoggerFactory
+
+class PluginCoreEnforcementActivity : ApplicationInitializedListener {
+    override suspend fun execute(asyncScope: CoroutineScope) {
+        // can't reference anything in core
+        val log = LoggerFactory.getLogger(this::class.java)
+
+        val coreId = PluginId.getId("aws.toolkit.core")
+        val coreDescriptor = PluginManagerCore.getPlugin(coreId)
+        if (coreDescriptor != null) {
+            // already installed
+            return
+        }
+
+        log.info("Attempting to install $coreId")
+        if (lookForPluginToInstall(coreId, EmptyProgressIndicator())) {
```

Comment:
they will be on 3.0 for this logic

Reply:
There's a relatively rare edge case that, after first IDE exit for toolkit update, it never restart. So 1 month later when user opens the IDE it will install a latest core that's unmatching what they have for toolkit(3.0).

Reply:
handler in core should handle that